iocore/net/YamlSNIConfig.cc

```diff
@@ -136,7 +136,7 @@ std::set<std::string> valid_sni_config_keys = {TS_fqdn,
                                                TS_client_sni_policy,
                                                TS_http2,
                                                TS_ip_allow,
-#if TS_USE_HELLO_CB
+#if TS_USE_HELLO_CB || defined(OPENSSL_IS_BORINGSSL)
                                                TS_valid_tls_versions_in,
```

Comment:
We may want to update the documentation.

https://github.com/apache/trafficserver/blame/3412c4f69552b10779cfe0ce1bc6a71971734491/doc/admin-guide/files/sni.yaml.en.rst#L99-L100